tests/network/localnet/conftest.py

```diff
@@ -54,33 +54,32 @@ def vlan_id(vlan_index_number: Generator[int]) -> int:
 
 
 @pytest.fixture(scope="module")
-def nad_localnet_1(
-    namespace_localnet_1: Namespace, vlan_id: int
-) -> Generator[netattachdef.NetworkAttachmentDefinition]:
-    with localnet_nad(namespace=namespace_localnet_1.name, name="test-localnet-nad1", vlan_id=vlan_id) as nad:
-        yield nad
+def cudn_localnet(
+    vlan_id: int,
+    namespace_localnet_1: Namespace,
+    namespace_localnet_2: Namespace,
+) -> Generator[libcudn.ClusterUserDefinedNetwork]:
+    with localnet_cudn(name=NETWORK_NAME, match_labels=LOCALNET_TEST_LABEL, vlan_id=vlan_id) as cudn:
+        cudn.wait_for_status_success()
+        yield cudn
 
 
 @pytest.fixture(scope="module")
-def nad_localnet_2(
-    namespace_localnet_2: Namespace, vlan_id: int
-) -> Generator[netattachdef.NetworkAttachmentDefinition]:
-    with localnet_nad(namespace=namespace_localnet_2.name, name="test-localnet-nad2", vlan_id=vlan_id) as nad:
-        yield nad
-
-
-@pytest.fixture(scope="module")
-def vm_localnet_1(nad_localnet_1: netattachdef.NetworkAttachmentDefinition) -> Generator[BaseVirtualMachine]:
+def vm_localnet_1(
+    namespace_localnet_1: Namespace, cudn_localnet: libcudn.ClusterUserDefinedNetwork
+) -> Generator[BaseVirtualMachine]:
     with localnet_vm(
-        namespace=nad_localnet_1.namespace, name="test-vm1", network=nad_localnet_1.name, cidr="10.0.0.1/24"
+        namespace=namespace_localnet_1.name, name="test-vm1", network=cudn_localnet.name, cidr="10.0.0.1/24"
```

Comment:
Maybe it would be beneficial to have a `cidr_subnet_prefix` var to ensure consistency between the  VMs.
But anyway this is not critical, it can be ignored/wait for a follow-up.

Reply:
This is not the scope of the current PR. The follow up added test will add this.